contrib/golang/filters/http/test/test_data/basic/filter.go

```diff
@@ -154,6 +163,15 @@ func (f *filter) encodeHeaders(header api.ResponseHeaderMap, endStream bool) api
 		return f.sendLocalReply("encode-header")
 	}
 	origin, _ := header.Get("x-test-header-0")
+	hdrs := header.Values("x-test-header-0")
+	if origin != "" {
+		if origin != hdrs[0] {
+			return f.fail("Values return incorrect data %v", hdrs)
+		}
+	} else if hdrs != nil {
+		return f.fail("Values return unexpected data %v", hdrs)
+	}
```

Comment:
There may be some headers with emtpy string values. It's possible that `origin == ""` and `hdrs != nil` both be true at same time.

Reply:
Thanks! Fixed in the new commit.